pkgs/cupertino_http/example/integration_test/url_session_task_test.dart

```diff
@@ -213,6 +213,9 @@ void testURLSessionTaskCommon(
           await request.drain<void>();
           request.response.headers.set('Content-Type', 'text/plain');
           request.response.write('Hello World');
+          // Add a small delay to allow cancellation to run before the request
+          // is completed.
```

Comment:
Is it feasible to coordinate the sequencing without an arbitrary delay? Should we wait on a `Completer` here and have the test body complete it after the cancellation runs? It will require edits in existing tests, but much better to be explicit about the interleaving.